gax/test/unit/streaming.ts

```diff
@@ -1466,6 +1474,67 @@ describe('streaming', () => {
       done();
     });
   });
+
+  it('properly emits the end event at the end of a pipeline transformation', done => {
+    const spy = sinon.spy((...args: Array<{}>) => {
+      assert.strictEqual(args.length, 3);
+      const s = new PassThrough({
+        objectMode: true,
+      });
+      s.push({resources: [1, 2]});
+      s.push(null);
+      setImmediate(() => {
+        s.emit('metadata');
+      });
+      return s;
+    });
+
+    // Initial stream.
+    const apiCall = createApiCallStreaming(
+      spy,
+      streaming.StreamType.SERVER_STREAMING
+    );
+    const s1 = apiCall({}, undefined);
+
+    // Transform stream.
+    const transform = new Transform({
+      objectMode: true,
+      transform: (data, _encoding, callback) => {
+        callback(
+          null,
+          data.resources.map((element: number) => element + 1)
+        );
+      },
+    });
+
+    // Final stream.
+    const s2 = new PassThrough({
+      objectMode: true,
+    });
+
+    const finalResults: Array<{resources: Array<number>}> = [];
+
+    s1.on('error', (err: GoogleError) => {
+      console.log('s1 error');
+      done(err);
+    });
+
+    s2.on('error', (err: GoogleError) => {
+      done(err);
+    });
+    s2.on('data', data => {
+      finalResults.push(data);
+    });
+    s2.on('end', () => {
+      assert.strictEqual(
+        JSON.stringify(finalResults),
+        JSON.stringify([[2, 3]])
+      );
+      done();
+    });
+
+    pipeline(s1, transform, s2, () => {});
```

Comment:
It would be worth adding error handling here along the lines of 
```suggestion
    pipeline(s1, transform, s2,  (err) => {
    if (err) {
      throw new Error('pipeline in properly emits the end event at the end of a pipeline transformation test failed')
    });
```